### DISCUSSION TITLE
Компоненты SimpleSwitch и Switch

### DISCUSSION THREAD

**Discussion by SevereCloud** [VKCOM/VKUI#2468] [Ideas]
Компонент [Switch](https://vkcom.github.io/VKUI/#/Switch) это надстройка над `<input type="checkbox" />`. Компонент принимает все валидные для этого элемента свойства.

Пример использования

```jsx
<SimpleCell
  role={null}
  disabled
  after={<Switch aria-label="Комментарии к записям" />}
>
  Комментарии к записям
</SimpleCell>
```

Нативное поведение на андроиде подразумевает, что нажатие на ячейку переключает с Switch. Тогда нам требуется следующий код:

```jsx
<SimpleCell
  Component="label"
  after={<Switch />}
>
  Комментарии к записям
</SimpleCell>
```

Но нативное поведение на ios другое - нажатие на ячейку не переключает Switch:

```jsx
<SimpleCell 
  Component={platform === IOS ? undefined : "label"} 
  disabled={platform === IOS} 
  after={<Switch aria-label="Комментарии к записям"/>}
>
  Комментарии к записям
</SimpleCell>
```

---

В v5 можно сделать два компонента:
- **SimpleSwitch** — обычный переключатель
- **Switch** — SimpleCell с SimpleSwitch, учитывающий нативное поведение

```jsx
<Switch>
  Комментарии к записям
</Switch>
```

FYI #2366

**Comment by eugpoloz** [comment]
> В v5 можно сделать два компонента:
>
> * **SimpleSwitch** — обычный переключатель
> * **Switch** — SimpleCell с SimpleSwitch, учитывающий нативное поведение

Вариант по названиям:
- оставить `Switch` как есть; 
- добавить `SwitchCell`/`CellSwitch` (= SimpleCell с SimpleSwitch, учитывающий нативное поведение).

Почему: у нас уже есть кнопка-ячейка `CellButton`, например; выглядеть будет более консистентно.

**Reply by inomdzhon** [reply]
`CellSwitch` найз 👍

**Comment by eugpoloz** [comment]
На всякий случай тегну еще дискуссию про базовые/simple-компоненты: https://github.com/VKCOM/VKUI/discussions/2362
